### ISSUE TITLE
Using C# to demonstrate Lock in Thread

### ISSUE THREAD

**Issue by Nganga89** [section-engineering-education/engineering-education#5427]
#### Proposed title of article
[Languages] Using C# to demonstrate Lock in Thread

#### Proposed article introduction
If another thread attempts to access the code currently being executed, the lock construct ensures that no other thread can access it. In multithreading programming, the other thread trying to enter the portion of code where an already existing thread is is made to wait or be blocked until the execution of the thread which is already executing in the piece of code completes. This lock is released once the current thread has finished its work in the part of code, enabling the other threads to continue working in the portion of code as needed.

#### Key takeaways 
By the end of this tutorial, the learner will be able to understand:
- What is meant with Lock in thread and what it entails in C# and moreso where to use it
- Functionality of the Lock feature in C#
- How to work with Lock in C#
- What are the advantages and disadvantages of this concept

#### Article quality
To make this article unique, I will explain all the concepts related to the topic in an in-depth manner. I will cover all possible approaches to this concept adding proper examples with precise outputs and comments. I will also include images wherever necessary. I will employ straightforward language to enhance understanding to make it more exciting and beginner-friendly. The learner will appreciate being subjected to problems with threads likely to arise while working with this lock concept, as I will still be covering it.

#### References
N/A

**Comment by LinusMuema** [comment]
Hi @Nganga89 

Could you please expand upon how your content will differ from what is already available through the [C# corner website](https://www.c-sharpcorner.com/UploadFile/de41d6/monitor-and-lock-in-C-Sharp/) and documentation? We lean towards content that offers an original viewpoint/approach around programming-centric topics.

**Comment by Nganga89** [comment]
Hello @LinusMuema I have checked the site that you suggested. In my opinion, I think it's impossible to exhaust knowledge and after going through almost similar blogs and articles I found that they are not beginner-friendly. Since an application's threads must cooperate and synchronize while sharing data and resources, I'll also address thread synchronization. Code and data locking are two of the most common ways to ensure that functions in modules are reentrant. Invariants play a vital role in limiting the complexity of locking in both code and data. This is one of the ways I hope to spice up the concept.
